cmd/rollout-operator/tls_server.go

```diff
@@ -9,31 +9,32 @@ import (
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
 
 	"github.com/grafana/rollout-operator/pkg/tlscert"
 )
 
 type tlsServer struct {
 	srv    *http.Server
-	mux    *http.ServeMux
+	mux    *mux.Router
 	port   int
 	logger log.Logger
 }
 
-func newTLSServer(port int, logger log.Logger, cert tlscert.Certificate) (*tlsServer, error) {
+func newTLSServer(port int, logger log.Logger, cert tlscert.Certificate, metrics *metrics) (*tlsServer, error) {
 	pair, err := tls.X509KeyPair(cert.Cert, cert.Key)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load key pair: %v", err)
 	}
 
-	mux := http.NewServeMux()
+	m, handler := newInstrumentedRouter(metrics)
```

Comment:
we could consider currying these metrics with a label which identifies the protocol as http or https, but that can also be done in a future PR so I'll approve this one

Reply:
We register different routes on the HTTP and HTTPS servers, so I don't think this is necessary.